src/panels/config/areas/ha-config-area-page.ts

```diff
@@ -246,13 +247,36 @@ class HaConfigAreaPage extends SubscribeMixin(LitElement) {
         .narrow=${this.narrow}
         .header=${area.name}
       >
-        <ha-icon-button
-          .path=${mdiPencil}
-          .entry=${area}
-          @click=${this._showSettings}
-          slot="toolbar-icon"
-          .label=${this.hass.localize("ui.panel.config.areas.edit_settings")}
-        ></ha-icon-button>
+        <ha-button-menu slot="toolbar-icon">
+          <ha-icon-button
+            slot="trigger"
+            .label=${this.hass.localize("ui.common.menu")}
+            .path=${mdiDotsVertical}
+          ></ha-icon-button>
+
+          <mwc-list-item
+            graphic="icon"
+            .entry=${area}
+            @click=${this._showSettings}
+          >
+            ${this.hass.localize("ui.panel.config.areas.edit_settings")}
+            <ha-svg-icon slot="graphic" .path=${mdiPencil}> </ha-svg-icon>
+          </mwc-list-item>
+
+          <mwc-list-item
+            class=${classMap({ warning: Boolean(this.areaId) })}
```

Comment:
I don't get this, there would always be an `areaId` right? Otherwise it makes no sense to show this item even?